src/nifti1.ts

```diff
@@ -0,0 +1,617 @@
+      //      [  0  -1   0 ]  which is "improper" (determinant = -1).
+      //      [  0   0   1 ]
+      //
+      //    If we set qfac=-1, then the R matrix would be
+      //
+      //      [  1   0   0 ]
+      //      [  0  -1   0 ]  which is proper.
+      //      [  0   0  -1 ]
+      //
+      //    This R matrix is represented by quaternion [a,b,c,d] = [0,1,0,0]
+      //    (which encodes a 180 degree rotation about the x-axis).
+
+      // Define a, b, c, d for coding covenience
+      const a = this.quatern_a;
+      const b = this.quatern_b;
+      const c = this.quatern_c;
+      const d = this.quatern_d;
+
+      this.qfac = this.pixDims[0] === 0 ? 1 : this.pixDims[0];
+
+      this.quatern_R = [
+        [
+          a * a + b * b - c * c - d * d,
+          2 * b * c - 2 * a * d,
+          2 * b * d + 2 * a * c,
+        ],
+        [
+          2 * b * c + 2 * a * d,
+          a * a + c * c - b * b - d * d,
+          2 * c * d - 2 * a * b,
+        ],
+        [
+          2 * b * d - 2 * a * c,
+          2 * c * d + 2 * a * b,
+          a * a + d * d - c * c - b * b,
+        ],
+      ];
+
+      for (ctrOut = 0; ctrOut < 3; ctrOut += 1) {
+        for (ctrIn = 0; ctrIn < 3; ctrIn += 1) {
+          this.affine[ctrOut]![ctrIn] =
+            this.quatern_R[ctrOut]![ctrIn]! * this.pixDims[ctrIn + 1]!;
+          if (ctrIn === 2) {
+            this.affine[ctrOut]![ctrIn] *= this.qfac!;
+          }
+        }
+      }
+      // The last row of affine matrix is the offset vector
+      this.affine[0][3] = this.qoffset_x;
+      this.affine[1][3] = this.qoffset_y;
+      this.affine[2][3] = this.qoffset_z;
+    } else if (this.sform_code > 0) {
+      //    METHOD 3 (used when sform_code > 0):
+      //    -----------------------------------
+      //    The (x,y,z) coordinates are given by a general affine transformation
+      //    of the (i,j,k) indexes:
+      //
+      //      x = srow_x[0] * i + srow_x[1] * j + srow_x[2] * k + srow_x[3]
+      //      y = srow_y[0] * i + srow_y[1] * j + srow_y[2] * k + srow_y[3]
+      //      z = srow_z[0] * i + srow_z[1] * j + srow_z[2] * k + srow_z[3]
+      //
+      //    The srow_* vectors are in the NIFTI_1 header.  Note that no use is
+      //    made of pixdim[] in this method.
+      for (ctrOut = 0; ctrOut < 3; ctrOut += 1) {
+        for (ctrIn = 0; ctrIn < 4; ctrIn += 1) {
+          index = 280 + (ctrOut * 4 + ctrIn) * 4;
+          this.affine[ctrOut]![ctrIn] = Utils.getFloatAt(
+            rawData,
+            index,
+            this.littleEndian
+          );
+        }
+      }
+    }
+
+    this.affine[3][0] = 0;
+    this.affine[3][1] = 0;
+    this.affine[3][2] = 0;
+    this.affine[3][3] = 1;
+
+    this.intent_name = Utils.getStringAt(rawData, 328, 344);
+    this.magic = Utils.getStringAt(rawData, 344, 348);
+
+    // TODO: typescript is catching that this comparison will always be false
+    this.isHDR = this.magic === NIFTI1.MAGIC_NUMBER2;
```

Comment:
TypeScript picks up on this comparison not making sense. Can't compare arrays (`MAGIC_NUMBER2`) like this in js.

Looks like `isHDR` isn't actually used anywhere so I suspect this code can be removed.